exp/internal/webhooks/machinepool_test.go

```diff
@@ -67,6 +69,169 @@ func TestMachinePoolDefault(t *testing.T) {
 	g.Expect(mp.Spec.Template.Spec.Version).To(Equal(ptr.To("v1.20.0")))
 }
 
+func TestCalculateMachinePoolReplicas(t *testing.T) {
+	tests := []struct {
+		name             string
+		newMP            *expv1.MachinePool
+		oldMP            *expv1.MachinePool
+		expectedReplicas int32
+		expectErr        bool
+	}{
+		{
+			name: "if new MP has replicas set, keep that value",
+			newMP: &expv1.MachinePool{
+				Spec: expv1.MachinePoolSpec{
+					Replicas: ptr.To[int32](5),
+				},
+			},
+			expectedReplicas: 5,
+		},
+		{
+			name:             "if new MP does not have replicas set and no annotations, use 1",
+			newMP:            &expv1.MachinePool{},
+			expectedReplicas: 1,
+		},
+		{
+			name: "if new MP only has min size annotation, fallback to 1",
+			newMP: &expv1.MachinePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						clusterv1.AutoscalerMinSizeAnnotation: "3",
+					},
+				},
+			},
+			expectedReplicas: 1,
```

Comment:
It would fall back to 1 by default unless both the min and max size annotations are set: https://github.com/willie-yao/cluster-api/blob/cc-mp-autoscaler/exp/internal/webhooks/machinepool.go#L288

This logic should be the same as for MDs: https://github.com/willie-yao/cluster-api/blob/cc-mp-autoscaler/internal/webhooks/machinedeployment.go#L385